### ISSUE TITLE
Finetuning SSD for video data

### ISSUE THREAD

**Issue by bhargavaurala** [weiliu89/caffe#862]
Hello,

I am trying to finetune the SSD model for a video dataset. Because it is a video I cannot use lmdb as the size would be too large. So I am planning to pass every frame to a caffe net object in python using code similar to examples/ssd/ssd_pascal_video.py and then using solver.step(1) to perform training.

Could you please let me know the correct preprocessing and format to pass the ground truth annotations of bounding boxes (I have annotations per frame) to the data layer so that we can compute multibox loss?

Please note I cannot use the create_dataset.sh because I have video dataset and not a list of images with corresponding ground truth files.

Bhargava

**Comment by ujsyehao** [comment]
what is your annotations per frame, what's the type of it?
image and video are essentially the same.

**Comment by bhargavaurala** [comment]
I have bounding box annotations per frame. In the training script examples/ssd_pascal.py, AnnotatedDataLayer is used with source as LMDB. Are there any other sources we can pass to this layer? Or are there layers I can use for this purpose?

**Comment by fraukej** [comment]
@bhargavaurala You can convert your annotations in xml, json or txt to LMDB using the supplied `create_data.sh` (changing the paths to your annotations/frames paths). You may need to execute `create_list.py` before.
